azurerm/resource_arm_devspace_controller.go

```diff
@@ -171,21 +171,19 @@ func resourceArmDevSpaceControllerRead(d *schema.ResourceData, meta interface{})
 		d.Set("location", azureRMNormalizeLocation(*location))
 	}
 
-	d.Set("sku", flattenDevSpaceControllerSku(result.Sku))
+	if err := d.Set("sku", flattenDevSpaceControllerSku(result.Sku)); err != nil {
+		return fmt.Errorf("Error flattenning Sku for DevSpace Controller %q (Resource Group %q): %+v", name, resGroupName, err)
+	}
 
+	var hostSuffix, dataPlaneFqdn, targetContainerHostResourceID *string
 	if props := result.ControllerProperties; props != nil {
-		if props.HostSuffix != nil {
-			d.Set("host_suffix", props.HostSuffix)
-		}
-
-		if props.DataPlaneFqdn != nil {
-			d.Set("data_plane_fqdn", props.DataPlaneFqdn)
-		}
-
-		if props.TargetContainerHostResourceID != nil {
-			d.Set("target_container_host_resource_id", props.TargetContainerHostResourceID)
-		}
+		hostSuffix = props.HostSuffix
+		dataPlaneFqdn = props.DataPlaneFqdn
+		targetContainerHostResourceID = props.TargetContainerHostResourceID
 	}
+	d.Set("host_suffix", hostSuffix)
+	d.Set("data_plane_fqdn", dataPlaneFqdn)
+	d.Set("target_container_host_resource_id", targetContainerHostResourceID)
```

Comment:
we can move these inside the `properties` block above rather than conditionally setting these; since these'll get caught by the import tests if they're unset

Reply:
Are you saying that `props` will never be `nil` in this case? If this is the case, can I just remove the `nil` check for props ?

Reply:
not quite - if `props` is `nil` then these properties won't be set as the code stands; which is the right approach, since these'll appear as an error on the Import tests - rather than showing as constant changes on users `terraform plan`'s (where there's not a whole bunch they can do to fix them)

Reply:
(this looks good with the changes - thanks 👍)